lua/zettelkasten.lua

```diff
@@ -58,8 +56,22 @@ local function get_all_tags(lookup_tag)
     return tags
 end
 
-local function generate_note_id()
-    return fn.strftime(NOTE_ID_STRFTIME_FORMAT)
+local function generate_note_id(parent_id)
+    local notes = browser.get_notes()
+
+    local id_format = config.get().id_format
+    local format_type = type(id_format)
+    if  format_type == 'string' then
+        return fn.strftime(id_format)
+    elseif  format_type == 'function' then
```

Comment:
```suggestion
    elseif format_type == 'function' then
```

Can you run `stylua` on your changes?